### ISSUE TITLE
Polaris Prep: Triage existing issues

### ISSUE THREAD

**Issue by MelSumner** [emberjs/ember.js#20473]
In order to get us in a better position to successfully ship Polaris features, we need to triage the existing issues. 

Evaluate existing issues: 
- try to reproduce the issue
- if there's a reproduction, try to do it on the most recent LTS of Ember

Guidance per project: 
- Ember Data: if the issue is open, it should be open (regularly triaged)
- Ember CLI: if it's older than 2 years, close the issue
- Ember JS: try to reproduce; 
- - if it's an old issue and it doesn't make sense, it's okay to close
- - if it seems to still be an issue, then give it a "still an issue" label
- - if it's a feature we no longer support (e.g. pods) then close the issue with "won't fix"

Phrasing to use when closing issues: 

"Unable to reproduce this issue; due to its age, closing issue. Please re-open with reproduction if you feel as though this is still an issue for you."

"This feature is no longer recommended or supported in Ember. Closing issue."

**Comment by sandstrom** [comment]
Great initiative @MelSumner! 

I worked through and closed ~30 issues a few weeks ago.

**Comment by johanrd** [comment]
Here's a small triage of **closing candidates**:

### Issue thread itself indicates resolved / cannot reproduce?

- #19421 — the issue body itself says fixed in 3.22.0; only affected the EOL 3.20 LTS.
- #17791 — "closed by [#19405](https://github.com/emberjs/ember.js/pull/19405)" per an in-thread contributor (merged 2021); never contested.
- #18446 — @kategengler (2025): "Is this still an issue?" — no repro ever provided, submitter silent since 2019.
- #19180 — the `logRenderStackForPath` machinery was removed entirely; could not reproduce (Feb 2026).

### The API or mechanism the report depends on no longer exists?

- #12945 — `Route#replaceWith` removed (RFC 674); same failure family as the open #18746.
- #18788 — paths with a tail are never helper-resolved anymore ([`isSimpleCallee`](https://github.com/emberjs/ember.js/blob/main/packages/@glimmer/syntax/lib/v2/loose-resolution.ts)); the issue's exact templates now give a clear build-time error instead.
- #19730 — curly `{{link-to}}` positional invocation was removed in 4.0; @chriskrycho in-thread: "doesn't work at all".
- #19763 — requests reverting intentional RFC 671 behavior; would need an RFC, not an issue.

### Seems fixed by a merged PR (but never (auto)closed)?

- #21145 — fixed by [`35438ae3bb`](https://github.com/emberjs/ember.js/commit/35438ae3bb); the commit message describes the exact retention problem.
- #20914 — fixed by [#20909](https://github.com/emberjs/ember.js/pull/20909) (merged 2025-06-05, regression test, shipped 6.8.0); auto-close never fired.
- #19821 — fixed by [`6416faf206`](https://github.com/emberjs/ember.js/commit/6416faf206); independently confirmed in-thread (2023).
- #20913 — ~the scope bag is still non-reactive on `main` (`template.ts` snapshots `options.scope?.()` once); the cited commits fixed `this`/keywords only.~ turned out **not** fixed Do not close.

**Comment by johanrd** [comment]
A second batch of **closing candidates** by Claude where close/reopen timelines checked, cited commits confirmed on `main`, threads read at source, reproductions attempted where feasible.

### Fixed / no longer reproduces?

- #15800 — fixed by [router.js#319](https://github.com/tildeio/router.js/pull/319) (QPs preserved through loading substates), shipped v3.27. Mechanism match; the commit cites #19307.
- #16266 — `RouterService.transitionTo` now uses the same `_doTransition` path the reporter confirmed abortable.
- #16719 — `{{each}}` docs now list the valid `key` values (`@index`, `@identity`).
- #18820 — the LinkTo rewrite removed the two-way `route` property the engines subclass collided with.
- #19015 — LinkTo `@query` is documented with a worked example.
- #19295 — `enumerableSymbol` no longer exists anywhere; premise gone.
- #19546 — LinkTo now binds click directly on the anchor; ancestor `stopPropagation` can no longer swallow it.
- #21029 — manager lookup now raises a clear assertion naming the offending definition.
- #21031 — verified by compile repro: errors carry module name + line/col. (Raw HBS parse errors still lack the module name.)

### Superseded — premise or mechanism no longer exists?

- #18132 — this-fallback removed in 4.0; the confusing implicit path is gone.
- #18710 — the optional features are no longer optional; the flags are gone from `main`.
- #18822 — classic v1-addon app-tree merging; superseded by v2 addons. (Classic builds may still repro.)
- #19074 — kategengler closed the community fix PR (2026): "in modern ember components are imported (in gjs/gts)".
- #19189 — depends on Ember Data 3.2x internals + render-modifiers identity; both replaced by autotracking.
- #19315 — glimmer-vm now lives in this repo (#20908); the two-repo docs premise is gone.
- #19401 — the colocation prepend is ember-cli-htmlbars territory (rwjblue); the RFC 977 blueprint has no prepend.
- #19432 — pre-3.25 AST tooling clash; resolved by addon upgrades.
- #19447 — fixed by [glimmer-vm#1276](https://github.com/glimmerjs/glimmer-vm/pull/1276); reporter acknowledged.
- #19476 — the monkey-patched `computed.macro` shims were removed in 4.0 ([`b005bc17eb`](https://github.com/emberjs/ember.js/commit/b005bc17eb)).
- #19724 — legacy AST plugin removal (#19429, v3.27); resolved addon-side.
- #19861 — kategengler: "We are very unlikely to change the behavior of `<LinkTo>` at this point."
- #19904 — ember-source ships a real `exports` map since 6.0; the rfc176-data premise is gone.

### Works as designed?

- #16867 — QPs are per route/controller; explained in-thread, unchallenged for 8 years.
- #16975 — identity keying over duplicate primitives (chancancode's analysis); the `{{action}}`-based repro is gone.
- #18090 — support question, no repro, silent for 7 years.
- #18118 — injections exist only after `setOwner`; reporter accepted (pzuraq).
- #18238 — the reporter's own framing is a design question; tightening QP declaration needs an RFC.
- #18321 — duplicate route names are a degenerate config; workaround in-thread.
- #18615 — `buildRouteInfoMetadata` may run before the controller exists, by design. (One-line docs note possible.)
- #18665 — router-service properties settle at `routeDidChange`, as documented.
- #18672 — QP aliasing works with the documented array form; verified in-thread.
- #18781 — WeakMaps/private fields break Proxies; platform limitation (pzuraq).
- #18795 — `destroyModifier` timing is deliberately loose per RFC 373; addon side tracked in ember-animated#202.
- #18808 — `transition.intent` is private; the supported surface is `transition.to`.
- #18873 — new-content-before-old-teardown is intended scheduled-destruction semantics.
- #18988 — autotracking short-circuit, diagnosed in-thread; never re-confirmed since.
- #19110 — the 3.22 backtracking assertion is intentional; the flagged addon patterns were the bug.
- #19222 — a raw `{{on "input"}}` inherently runs before `<Input>`'s two-way write-back; docs guidance exists.
- #19265 — the tabs come from this repo's own `target=_blank` LinkTo tests; skip toggle added in [`e4af7b204a`](https://github.com/emberjs/ember.js/commit/e4af7b204a).
- #19268 — `fn` deliberately returns a stable identity that reads the latest args.
- #19503 — named blocks must be direct children, by design; the feature ask lives in rfcs#735.
- #19753 — build-time deprecations bypass the runtime workflow; documented limitation, owned elsewhere.
- #19758 — helper-manager args are a stable lazy proxy by design.
- #19794 — scoped service lookup is expected per [ember-resolver#531](https://github.com/ember-cli/ember-resolver/pull/531).
- #20632 — value-equality dedupe is a userland concern (tracked-toolbox has it).
- #21032 — the capability has no consumer: `didCreateElement` only fires for wrapped classic components.

### Verified by fresh reproduction attempts (2026-07-27)

- #11288 — wrote the missing test: a `serializeQueryParam` override IS used on `main`; the report does not reproduce.
- #17743 — reconstructed the scenario (retrying an aborted initial transition): no crash on `main`.
- #18608 — answered, with tests both ways: explicit `this.route('index')` defaults to path `/index`, silently unmatching the parent URL; `this.route('index', { path: '/' })` works. Possible docs follow-up.
- #19332 — premise removed: only `'3.22'` modifier capabilities exist on `main`; `'3.13'` asserts at registration.


### Belongs in a different repository?

- #18513 — new-public-API request → emberjs/rfcs.
- #18929 — guides content; already transferred (guides-source#1443, closed).
- #18959 — watchman/EMFILE detection → ember-cli.
- #19239 — Chromium BFCache behavior, by the author's own framing.
- #19507 — serializer/model blueprints ship from ember-data.

### Duplicates of live issues?

- #10557, #14880, #14929 — the LinkTo active-class-after-retry family; canonical is the open #15518 (has a 5.5 repro).
- #19679 — duplicate of the open #19573 (`transitionTo` returned from async hooks hangs).

### Obsolete by time?

- #19033 — 3.16–3.19 leak; fix confirmed in-thread on 3.20.2.
- #19702 — time-boxed Ember 4.0 docs epic, long expired.
- #19240 — reproduced on `main`: native `length`/`some` on an EmberArray do NOT autotrack. Yet, with RFC 1114 (Deprecate Ember Array) accepted, a tracking fix seems unlikely — probably close + a guides-source note, like #19287.
- #20628 — size delta between two EOL releases (5.5 → 5.6), but @NullVoxPopuli's quest for bundle size reduction is heavily ongoing it seems💪

**Comment by NullVoxPopuli** [comment]
@johanrd I was reminded that I made this a while back that tracks open vs closed issues in various ways:

https://is-polaris-ready.nullvoxpopuli.com/issues

it could also be used for triage, as a lot of things that are white squares are actually kinda old ( oops )
